### PR TITLE
Detect when isolate reload is barred and inform the user

### PR DESCRIPTION
**Pull request by johnmccutchan**
- [x] React to the signal from the VM that an isolate is barred from performing a hot reload and must be restarted.
- [x] Roll the engine forward

### CODE REVIEW

**Comment by johnmccutchan** [comment]
@devoncarew @Hixie

**Comment by devoncarew** [comment]
I think some of the engine artifacts aren't yet available on google storage.

File: packages/flutter_tools/lib/src/hot.dart [hunk 1]
```diff
@@ -445,9 +445,18 @@ class HotRunner extends ResidentRunner {
       } else {
         flutterUsage.sendEvent('hot', 'reload');
       }
-    } catch (errorMessage, st) {
+    } catch (error, st) {
```
Comment (devoncarew) [inline packages/flutter_tools/lib/src/hot.dart]: Should we ensure that we're only catching a `Map` here (`on Map`)?

**Comment by devoncarew** [comment]
lgtm, but you probably want to wait to land this until all the engine artifacts build through.

**Comment by yjbanov** [comment]
A lot of breakages caused by this PR ([dashboard](https://flutter-dashboard.appspot.com/build.html), [sample log file](https://flutter-dashboard.appspot.com/api/get-log?ownerKey=ahNzfmZsdXR0ZXItZGFzaGJvYXJkclgLEglDaGVja2xpc3QiOGZsdXR0ZXIvZmx1dHRlci8wOWU2OGMzM2NjZGIzNTVmZWRkNzVhODcxNmRlMDU1MWM3NzJmMmRlDAsSBFRhc2sYgICAgICAwAgM))

**Comment by yjbanov** [comment]
Rolling it back in https://github.com/flutter/flutter/pull/5590

**Comment by johnmccutchan** [comment]
Thanks Yegor!

**Pull request merged**